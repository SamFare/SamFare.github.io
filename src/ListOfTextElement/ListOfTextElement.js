import React from "react";
import PropTypes from "prop-types";
import TextElement from "../TextElement/TextElement.js";
import InvertedTextElement from "../TextElement/InvertedTextElement.js";

export default class ListOfTextElement extends React.Component{
  
  constructor(props) {
    super(props);
    const { elements } = this.props;
    this.listItems = this.generateElements(elements); 
    this.generateElements = this.generateElements.bind(this);
  }

  static get propTypes() { 
    return {
      elements: PropTypes.array,
    };
  }

  generateElements(elementJson) {
    let index = 0;
    let elementAsJsx = []; 
    
    for(let element of elementJson) {
      if(index % 2) {
        elementAsJsx.push(<InvertedTextElement key={index} img={element.img}/>);
      } else {
        elementAsJsx.push(<TextElement key={index} />);
      }
      ++index;
    }

    return elementAsJsx;
  }

  render(){
    return (
      <div>
        {this.listItems}
      </div>
    );
  } 
}
