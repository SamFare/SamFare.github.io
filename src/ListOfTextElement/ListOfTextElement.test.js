import React from "react";
import ListOfTextElement from "./ListOfTextElement.js";
import renderer from "react-test-renderer";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

let context = {};

describe("there are three elements ",  () => {
  beforeEach(() => {
    context.elementArray =  [
      {img: "test.png", text: "this is text" },
      {img: "test2.png", text: "this is text" },
      {img: "test3.png", text: "this is text" }
    ];
  
    context.listOfTextElement = renderer.create(
      <ListOfTextElement elements={context.elementArray} />
    ); 
  });

  test("has the number of elemebts that is requested",() => {
    expect(context.listOfTextElement.getInstance().listItems.length).toEqual(3);
  });

  test("the elements have the correct text passed to them",() => {
    expect(context.listOfTextElement.getInstance().listItems[0].props.text).toEqual("this is text");
  });

  test("the elements have the correct image passed to them",() => {
    expect(context.listOfTextElement.getInstance().listItems[0].props.img).toEqual("test.png");
  });

  test("matches the snapshot",() => {
    expect(context.listOfTextElement).toMatchSnapshot();
  });
});

describe("there are no elements ",  () => {
  beforeEach(() => {
    context.listOfTextElement = renderer.create(
      <ListOfTextElement elements={[]} />
    );
  });

  test("has the number of elements that is requested",() => {
    expect(context.listOfTextElement.getInstance().listItems.length).toEqual(0);
  });
});
