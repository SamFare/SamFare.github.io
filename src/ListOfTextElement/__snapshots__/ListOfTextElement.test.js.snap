// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`there are three elements  matches the snapshot 1`] = `
<div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <img
        height="100"
        src="this is text"
        width="100"
      />
    </div>
    <div
      className={undefined}
    >
      test.png
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam porta accumsan molestie. Suspendisse potenti. Vivamus eu tempus dui, eu consectetur metus. Sed non est gravida, egestas est sed, vulputate tortor. Sed quis eros quis elit congue tempor vitae vitae enim. Aliquam sed ultrices elit. Duis venenatis consequat enim, faucibus porttitor eros suscipit non. Pellentesque rutrum aliquet nibh, eget iaculis nibh lobortis bibendum. Proin consectetur, turpis eget blandit ultricies, neque nisi facilisis leo, vel dapibus velit est dignissim nunc.
    </div>
    <div
      className={undefined}
    >
      <img
        height="100"
        src="../../assets/about-page-pic.jpg"
        width="100"
      />
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <img
        height="100"
        src="this is text"
        width="100"
      />
    </div>
    <div
      className={undefined}
    >
      test3.png
    </div>
  </div>
</div>
`;
